cmd/antrea-agent/agent.go

```diff
@@ -182,6 +189,10 @@ func run(o *Options) error {
 
 	go agentMonitor.Run(stopCh)
 
+	if features.DefaultFeatureGate.Enabled(features.AntreaProxy) && proxier != nil {
```

Comment:
I think either condition is fine but no need to have two conditions

Reply:
You're right, this is used to satisfy my local nil pointer checker.
Because of the short path evaluation of boolean expression, I think the last expression will not take computation.

Reply:
I will update later because there is an on-going test and I want to check the result.

Reply:
Updated.